app/forms/developers_search_form.rb

```diff
@@ -0,0 +1,18 @@
+class DevelopersSearchForm
+  include Virtus.model
+  include ActiveModel::Model
+  include ActiveRecord::Sanitization::ClassMethods
+
+  attribute :programming_language_id, String
+  attribute :language_id, String
+
+  PARAMS = [:programming_language_id, :language_id].freeze
+
+  def search
+    query = Developer.all.includes(:developer_programming_languages, :programming_languages, :developer_languages, :languages)
```

Comment:
@tooooomin `.all` not necessary

Reply:
Do you need to includes all of them? (This is a question. So please check the code and log yourself)
Unnecessary `includes` bring some performance issue. (Most of the time, it's not a big problem, though)